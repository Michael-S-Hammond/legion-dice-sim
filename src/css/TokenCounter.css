.token-counter-aim {
  background-image: url("../img/aim.png");
  display: flex;
  width: 40px;
  height: 40px;
  background-repeat: no-repeat;
  background-size: 85%;
  background-position-x: 2px;
  background-position-y: 2px; }

.token-counter-dodge {
  background-image: url("../img/Dodge.png");
  display: flex;
  width: 40px;
  height: 40px;
  background-repeat: no-repeat;
  background-size: 85%;
  background-position-x: 2px;
  background-position-y: 2px; }

.token-counter-observation {
  background-image: url("../img/Observation.png");
  display: flex;
  width: 40px;
  height: 40px;
  background-repeat: no-repeat;
  background-size: 85%;
  background-position-x: 2px;
  background-position-y: 2px; }

.token-counter-shield {
  background-image: url("../img/Shield.png");
  display: flex;
  width: 40px;
  height: 40px;
  background-repeat: no-repeat;
  background-size: 85%;
  background-position-x: 2px;
  background-position-y: 2px; }

.token-counter-suppression {
  background-image: url("../img/suppression.png");
  display: flex;
  width: 40px;
  height: 40px;
  background-repeat: no-repeat;
  background-size: 85%;
  background-position-x: 2px;
  background-position-y: 2px; }

.token-counter-surge {
  background-image: url("../img/surge.png");
  display: flex;
  width: 40px;
  height: 40px;
  background-repeat: no-repeat;
  background-size: 85%;
  background-position-x: 2px;
  background-position-y: 2px; }

.token-parent {
  position: relative; }

.token-count {
  font-weight: bold;
  text-align: center;
  font-size: small;
  border-radius: 50%;
  width: 20px;
  height: 20px;
  background-color: white;
  position: absolute;
  bottom: 1px;
  right: 5px;
  pointer-events: none; }
