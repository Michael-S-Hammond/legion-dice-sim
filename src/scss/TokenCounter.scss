@mixin token-counter {
    display: flex;
    width: 40px;
    height: 40px;
    background-repeat: no-repeat;
    background-size: 85%;
    background-position-x: 2px;
    background-position-y: 2px;
}

.token-counter-aim {
    background-image: url('../img/aim.png');
    @include token-counter();
}

.token-counter-dodge {
    background-image: url('../img/Dodge.png');
    @include token-counter();
}

.token-counter-observation {
    background-image: url('../img/Observation.png');
    @include token-counter();
}

.token-counter-shield {
    background-image: url('../img/Shield.png');
    @include token-counter();
}

.token-counter-suppression {
    background-image: url('../img/suppression.png');
    @include token-counter();
}

.token-counter-surge {
    background-image: url('../img/surge.png');
    @include token-counter();
}

.token-parent {
    position: relative;
}

.token-count {
    font-weight: bold;
    text-align: center;
    font-size: small;
    border-radius: 50%;
    width: 20px;
    height: 20px;
    background-color: white;
    position: absolute;
    bottom: 1px;
    right: 5px;
    pointer-events: none;
}
