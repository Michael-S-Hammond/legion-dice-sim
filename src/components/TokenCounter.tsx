import '../css/TokenCounter.css';

import React from 'react';

type TokenProps = {
    visible: boolean,
    value: number,
    tokenCssClass: string,
    onClick: () => void,
    tooltip: string,
}

class Token extends React.Component<TokenProps> {
    constructor(props : TokenProps) {
        super(props);
    }

    render() : JSX.Element {
        return (
            <div className={`${ this.props.visible ? 'collapse.show' : 'collapse'} token-parent`}>
                <button
                    className={`btn border btn-light border-secondary rounded-lg mx-1 ${this.props.tokenCssClass}`}
                    onClick={() => this.props.onClick()}
                    title={this.props.tooltip}>
                </button>
                <span className="token-count">{this.props.value}</span>
            </div>
        );
    }
}

export default Token;
