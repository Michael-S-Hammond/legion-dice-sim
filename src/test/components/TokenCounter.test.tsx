import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';

import TokenCounter from '../../components/TokenCounter';

describe('TokenCounter', () => {
    it('matches the snapshot', () => {
        function onClick() {
        }

        const counter = shallow(<TokenCounter
            visible={true}
            value={3}
            onClick={onClick}
            tokenCssClass='token-counter-aim'
            tooltip='Helpful text'
            ></TokenCounter>);
        const snapshot = shallowToJson(counter);
        expect(snapshot).toMatchSnapshot();
    });

    it('matches the snapshot when not visible', () => {
        function onClick() {
        }

        const counter = shallow(<TokenCounter
            visible={false}
            value={1}
            onClick={onClick}
            tokenCssClass='token-counter-observe'
            tooltip='Watching!'
            ></TokenCounter>);
        const snapshot = shallowToJson(counter);
        expect(snapshot).toMatchSnapshot();
    });

    it('handles being clicked', () => {
        const onButtonClickMock = jest.fn();

        let wrapper = shallow(<TokenCounter
            visible={true}
            value={2}
            onClick={onButtonClickMock}
            tokenCssClass='token-counter-dodge'
            tooltip='Dodging...'
            ></TokenCounter>);
        wrapper.find('.btn').simulate('click');

        expect(onButtonClickMock).toHaveBeenCalledTimes(1);
    })
});
