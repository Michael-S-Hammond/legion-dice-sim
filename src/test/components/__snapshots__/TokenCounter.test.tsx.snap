// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TokenCounter matches the snapshot 1`] = `
<div
  className="collapse.show token-parent"
>
  <button
    className="btn border btn-light border-secondary rounded-lg mx-1 token-counter-aim"
    onClick={[Function]}
    title="Helpful text"
  />
  <span
    className="token-count"
  >
    3
  </span>
</div>
`;

exports[`TokenCounter matches the snapshot when not visible 1`] = `
<div
  className="collapse token-parent"
>
  <button
    className="btn border btn-light border-secondary rounded-lg mx-1 token-counter-observe"
    onClick={[Function]}
    title="Watching!"
  />
  <span
    className="token-count"
  >
    1
  </span>
</div>
`;
